File: Day01/day1.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.*;

public class day1 {
    public static void main(String[] args) {
        ArrayList<Integer> elfCalories = new ArrayList<>();

        try {
            BufferedReader reader = new BufferedReader(new FileReader("input.txt"));

            int caloriesSum = 0;

            String input;
            while ((input = reader.readLine()) != null) {
                if (!input.equals("")) {
                    caloriesSum += Integer.parseInt(input);
                } else {
                    elfCalories.add(caloriesSum);
                    caloriesSum = 0;
                }
            }
            elfCalories.add(caloriesSum);
            reader.close();
        } catch (IOException e) {
            e.printStackTrace();
        }

        Collections.sort(elfCalories, Collections.reverseOrder());

        // Part 1 Answer
        System.out.println(elfCalories.get(0));

        // Part 2 Answer
        System.out.println(elfCalories.get(0) + elfCalories.get(1) + elfCalories.get(2));
    }
}

File: Day01/day1.py
with open("input.txt") as f:
    calories_sum = []
    current_sum = 0
    for calories in f:
        calories = calories.strip("\n")
        if calories:
            current_sum += int(calories)
        else:
            calories_sum.append(current_sum)
            current_sum = 0
    calories_sum.sort(reverse=True)

    # Part 1 Answer
    print(calories_sum[0])

    # Part 2 Answer
    print(sum(calories_sum[:3]))